ebpfsvc/svcmain.cpp

```diff
@@ -73,33 +75,48 @@ service_install()
         SC_MANAGER_ALL_ACCESS); // full access rights
 
     if (nullptr == scmanager) {
-        return;
+        result = GetLastError();
+        goto Exit;
     }
 
-    // Create the service
+    // Create the service as LocalService.
 
     service = CreateService(
-        scmanager,                 // SCM database
-        SERVICE_NAME,              // name of service
-        SERVICE_NAME,              // service name to display
-        SERVICE_ALL_ACCESS,        // desired access
-        SERVICE_WIN32_OWN_PROCESS, // service type
-        SERVICE_DEMAND_START,      // start type
-        SERVICE_ERROR_NORMAL,      // error control type
-        path,                      // path to service's binary
-        nullptr,                   // no load ordering group
-        nullptr,                   // no tag identifier
-        nullptr,                   // no dependencies
-        nullptr,                   // LocalSystem account
-        nullptr);                  // no password
+        scmanager,                     // SCM database
+        SERVICE_NAME,                  // name of service
+        SERVICE_NAME,                  // service name to display
+        SERVICE_ALL_ACCESS,            // desired access
+        SERVICE_WIN32_OWN_PROCESS,     // service type
+        SERVICE_AUTO_START,            // start type
+        SERVICE_ERROR_NORMAL,          // error control type
+        path,                          // path to service's binary
+        nullptr,                       // no load ordering group
+        nullptr,                       // no tag identifier
+        nullptr,                       // no dependencies
+        L"NT AUTHORITY\\LocalService", // LocalService account
+        nullptr);                      // no password
 
     if (service == nullptr) {
+        result = GetLastError();
+        goto Exit;
+    }
+
+    // Set service SID type to unrestricted.
```

Comment:
I couldn't find useful public docs on restricted on unrestricted.  Can the comment explain why unrestricted rather than restricted is the correct value here?

Reply:
Tested that restricted is sufficient here. Changed sidtype to restricted.